### DISCUSSION TITLE
In RDKitJS does one need to delete all the molecules inside a JSMolList

### DISCUSSION THREAD

**Discussion by syedzayyan** [rdkit/rdkit#7350] [Development]
Hello!

I am working with JSMolList and was wondering if doing .delete() was enough to deallocate the memory on the list or if it was necessary to delete individual items with a loop? I am passing the JSMolList to another RDKitJS function and not really accessing the molecules within.

**Comment by ptosco** [comment]
@syedzayyan Calling `MolList.delete()` is sufficient, as the molecules it contains are `ROMOL_SPTR` which live in the WASM layer, and they will be freed when you call `MolList.delete()`. However, you will need to call `delete()` on each `Mol` object that you retrieve from the WASM layer into the JS layer from `MolList` calling `MolList.at()` or `MolList.next()`.

**Reply by syedzayyan** [reply]
Thank you!
